src/index.ts

```diff
@@ -20,72 +20,135 @@
-const PORT = 3001;
-const iriBase = `http://localhost:${PORT}`;
-const storage = new InMemoryStorage();
+type InputOptions = {
+  solidIdentityProvider?: string;
+  clientName?: string;
+  registrationType?: "static" | "dynamic";
+};
+
+type ValidatedOptions = {
+  solidIdentityProvider: string;
+  registrationType: "static" | "dynamic";
+  clientName?: string;
+};
 
-// Initialised when the server comes up and is running...
-let session: Session;
+async function main(): Promise<void> {
+  // Get CLI arguments
+  const argv = require("yargs/yargs")(process.argv.slice(2))
+    .describe(
+      "solidIdentityProvider",
+      "The identity provider at which the user should authenticate."
+    )
+    .alias("idp", "solidIdentityProvider")
+    .describe("appName", "The name of the app you are registering.")
+    .describe(
+      "registration",
+      "[static] if you want to manually register the client, [dynamic] otherwise."
+    )
+    .describe(
+      "port",
+      "The port number on which the identity provider will return the token."
+    )
+    .locale("en")
+    .help().argv;
 
-const server = app.listen(PORT, async () => {
-  session = new Session({
+  const inputOptions: InputOptions = {
+    ...argv,
+    clientName: argv.appName,
+  };
+  // Complete CLI arguments with user prompt
+  const validatedOptions: ValidatedOptions = {
+    solidIdentityProvider:
+      inputOptions.solidIdentityProvider ?? (await promptIdp()),
+    registrationType:
+      inputOptions.registrationType ?? (await promptRegistration()),
+    clientName: inputOptions.clientName ?? (await promptClientName()),
+  };
+  const port = argv.port ?? (await promptPort());
+
+  const app = express();
+  const iriBase = `http://localhost:${port}`;
+  const storage = new InMemoryStorage();
+
+  const session: Session = new Session({
     insecureStorage: storage,
     secureStorage: storage,
   });
 
-  console.log(`Listening at: [http://localhost:${PORT}].`);
-  console.log(`Logging in ${argv.oidcIssuer} to get a refresh token.`);
-  session
-    .login({
-      clientName: argv.clientName,
-      oidcIssuer: argv.oidcIssuer,
+  const server = app.listen(port, async () => {
+    console.log(`Listening at: [${iriBase}].`);
+    const loginOptions: ILoginInputOptions = {
+      clientName: validatedOptions.clientName,
+      oidcIssuer: validatedOptions.solidIdentityProvider,
       redirectUrl: iriBase,
       tokenType: "DPoP",
       handleRedirect: (url) => {
         console.log(`\nPlease visit ${url} in a web browser.\n`);
       },
-    })
-    .catch((e) => {
+    };
+    let clientInfo;
+    if (validatedOptions.registrationType === "static") {
+      console.log(
+        `Please go perform the static registration of your application to [${validatedOptions.solidIdentityProvider}].`
+      );
+      console.log(`The redirect IRI will be ${iriBase}`);
+      console.log(
+        "At the end of the registration process, you should get a client ID and secret."
```

Comment:
```suggestion
        "At the end of the static registration process, you should get a client ID and secret (which you can then provide to this application using the [$param-client-id}] and [${param-client-secret}] command line options."
```
(I don't know if you have constants for `param-client-id` and `param-client-secret`, but if not I'd add them so that you can refer to them here (as opposed to hard-coding them directly into this template string)).

Reply:
Actually, you won't need to pass the client ID and secret as CLI options if you reach this point. We could add them as CLI options later so that you're not prompted for them here. But basically, once you gave that to `generate-oidc-token`, it can go and use them at the token endpoint, so you don't have to re-run the script passing them as options.